src/libsyntax/feature_gate.rs

```diff
@@ -1284,16 +1284,15 @@ impl<'a> Context<'a> {
                 return;
             }
         }
-        if name.starts_with("rustc_") {
-            gate_feature!(self, rustc_attrs, attr.span,
-                          "unless otherwise specified, attributes \
-                           with the prefix `rustc_` \
-                           are reserved for internal compiler diagnostics");
-        } else if !attr::is_known(attr) {
-            // Only run the custom attribute lint during regular feature gate
-            // checking. Macro gating runs before the plugin attributes are
-            // registered, so we skip this in that case.
-            if !is_macro {
+        if !attr::is_known(attr) {
+            if name.starts_with("rustc_") {
+                let msg = "unless otherwise specified, attributes with the prefix `rustc_` \
+                           are reserved for internal compiler diagnostics";
+                gate_feature!(self, rustc_attrs, attr.span, msg);
+            } else if !is_macro {
+                // Only run the custom attribute lint during regular feature gate
+                // checking. Macro gating runs before the plugin attributes are
+                // registered, so we skip this in that case.
```

Comment:
Was this change _needed_? I don't mind it, just curious if it changed anything.

Reply:
Yes, it's to get rid of duplicate ``attributes with the prefix `rustc_` are reserved`` errors.

Reply:
Similarly to few other changes in https://github.com/rust-lang/rust/pull/56999/commits/21370386f370c828fe5378387ee7223a931e1480.
The diff looked pretty bad when I *just* removed the `abort_if_error`, all the other stuff is attempts to reduce noise from duplicated or nonsensical errors.